src/Plugin.php

```diff
@@ -33,4 +39,112 @@ class Plugin extends BasePlugin
      * @var bool
      */
     protected $routesEnabled = false;
+
+    /**
+     * Add migrations commands.
+     *
+     * @param \Cake\Console\CommandCollection $collection The command collection to update
+     * @return \Cake\Console\CommandCollection
+     */
+    public function console(CommandCollection $collection): CommandCollection
+    {
+        if (class_exists('Bake\Command\SimpleBakeCommand')) {
+            $commands = $collection->discoverPlugin($this->getName());
```

Comment:
Would it be possible to always use `discoverPlugin()` and when bake is not present unset the commands that depend on Bake? That might let you avoid duplicating code from the command scanner.

Reply:
unfortunately using `discoverPlugin` causes the error

I explained here:
https://github.com/cakephp/migrations/issues/433#issuecomment-593250034

Reply:
I can add them manually
what do you think?

Reply:
Adding commands manually is how I solved this in authorization as well.

Reply:
done.